pkg/scaling/resolver/aws_secretManager_handler.go

```diff
@@ -0,0 +1,120 @@
+package resolver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+)
+
+type AwsSecretManagerHandler struct {
+	secretManager *kedav1alpha1.AwsSecretManager
+	session       *secretsmanager.Client
+}
+
+func NewAwsSecretManagerHandler(a *kedav1alpha1.AwsSecretManager) *AwsSecretManagerHandler {
+	return &AwsSecretManagerHandler{
+		secretManager: a,
+	}
+}
+
+func (ash *AwsSecretManagerHandler) Read(ctx context.Context, logger logr.Logger, secretName, versionID, versionStage string) (string, error) {
+	input := &secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(secretName),
+	}
+	if versionID != "" {
+		input.VersionId = aws.String(versionID)
+	}
+	if versionStage != "" {
+		input.VersionStage = aws.String(versionStage)
+	}
+	result, err := ash.session.GetSecretValue(ctx, input)
+	if err != nil {
+		logger.Error(err, "Error getting credentials")
+		return "", err
+	}
+	return *result.SecretString, nil
+}
+
+func (ash *AwsSecretManagerHandler) Initialize(ctx context.Context, client client.Client, logger logr.Logger, triggerNamespace string, secretsLister corev1listers.SecretLister, podTemplateSpec *corev1.PodTemplateSpec) error {
+	config, err := ash.getcredentials(ctx, client, logger, triggerNamespace, secretsLister, podTemplateSpec)
+	if err != nil {
+		logger.Error(err, "Error getting credentials")
+		return err
+	}
+
+	logger.Info("Config value", "config", config)
+
+	if ash.secretManager.Cloud != nil {
+		if ash.secretManager.Cloud.Region != "" {
+			config.Region = ash.secretManager.Cloud.Region
+		}
+		if ash.secretManager.Cloud.Endpoint != "" {
+			logger.Info("Endpoint value", "Endpoint", ash.secretManager.Cloud.Endpoint)
+		}
```

Comment:
Shouldn't we have to pass `ash.secretManager.Cloud.Endpoint` to config?

Reply:
(and please, remove the log or set V(1) to filter log level

Reply:
Am yet to figure out the logic in aws-go-sdk-v2.
I used the log for temporarily code